src/main/kotlin/net/atos/client/zgw/ztc/ZTCClientService.kt

```diff
@@ -55,6 +56,58 @@ class ZTCClientService : Caching {
     @Inject
     private lateinit var configuratieService: ConfiguratieService
 
+    companion object {
+        private val CACHES = listOf(
+            Caching.ZTC_BESLUITTYPE,
+            Caching.ZTC_CACHE_TIME,
+            Caching.ZTC_RESULTAATTYPE,
+            Caching.ZTC_STATUSTYPE,
+            Caching.ZTC_INFORMATIEOBJECTTYPE,
+            Caching.ZTC_ZAAKTYPE_INFORMATIEOBJECTTYPE,
+            Caching.ZTC_ZAAKTYPE
+        )
+
+        private val LOG = Logger.getLogger(ZTCClientService::class.java.name)
+
+        private fun <K, V> createCache(size: Long = 100): Cache<K, V> =
+            Caffeine.newBuilder()
+                .maximumSize(size)
+                .recordStats()
+                .removalListener { key: K?, _: V?, cause ->
+                    LOG.info("Remove key : $key because : $cause")
+                }.build()
+
+        private val ztcTimeCache: Cache<String, ZonedDateTime> = createCache(1)
+
+        private val uuidToZaakTypeCache: Cache<UUID, ZaakType> = createCache()
```

Comment:
You don't want to combine all zaaktype caches into one just yet? We could do that in the next iteration but I think it's quite simple:

```private val zaakTypeCache: Cache<Any, ZaakType> = createCache()```

Reply:
same for the other caches

Reply:
> You don't want to combine all zaaktype caches into one just yet? We could do that in the next iteration but I think it's quite simple:
> 
> ```private val zaakTypeCache: Cache<Any, ZaakType> = createCache()```

I wanted to have clean statistics for every endpoint + typed checks on params

We don't lose too much space in this way, except the object references

Reply:
hmm yeah ok. But we do now create more caches with more memory consumption than we need possibly and also this makes our code a bit more brittle because when invalidating caches you need to make sure you invalidate all individual caches explicitly. But yeah, typed checks and more detailed stats are also nice to have so let's go for this indeed.